### PR TITLE
Remove unreachable code

### PR DESCRIPTION
**Pull request by sharanry**
https://github.com/SciML/SciMLBase.jl/issues/83

### CODE REVIEW

File: src/vector_of_array.jl [hunk 1]
```diff
@@ -72,11 +72,7 @@ Base.@propagate_inbounds function Base.getindex(A::AbstractDiffEqArray{T, N},sym
   end
 
   if i === nothing
-    if issymbollike(i) && A.indepsym !== nothing && Symbol(i) == A.indepsym
-      A.t
```
Comment (ChrisRackauckas) [inline src/vector_of_array.jl:76]: did you try the independent variable and see what it does?

Comment (sharanry) [reply src/vector_of_array.jl:76]: I think that case is covered now. Added tests too.

**Review by ChrisRackauckas — APPROVED** [review]

File: src/vector_of_array.jl [hunk 2]
```diff
@@ -71,8 +71,8 @@ Base.@propagate_inbounds function Base.getindex(A::AbstractDiffEqArray{T, N},sym
     i = sym
   end
 
-  if i === nothing
-    if issymbollike(i) && A.indepsym !== nothing && Symbol(i) == A.indepsym
+  if i == nothing
```
Comment (YingboMa) [inline src/vector_of_array.jl:74]: This is not recommended. Why do you need to use `==`?

**Review by YingboMa — COMMENTED** [review]
Do you need `==`'s generic methods?

File: src/vector_of_array.jl (hunk 2 above)
Comment (sharanry) [reply src/vector_of_array.jl:74]: Please see https://github.com/SciML/SciMLBase.jl/pull/84#discussion_r667998159 .

**Comment by sharanry** [comment]
> Do you need `==`'s generic methods?

Are you referring to the same `==`/`===` issue?

**Pull request merged**